nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -555,9 +573,15 @@ in {
     services.postfix = lib.mkIf (cfg.smtp.createLocally && cfg.smtp.host == "127.0.0.1") {
       enable = true;
     };
-    services.redis = lib.mkIf (cfg.redis.createLocally && cfg.redis.host == "127.0.0.1") {
-      enable = true;
-    };
+    services.redis = lib.mkIf (cfg.redis.createLocally && cfg.redis.host == "127.0.0.1") (lib.mkMerge [
+      ({
+        enable = true;
+      })
+      (lib.mkIf cfg.redis.enableUnixSocket {
+        unixSocket = "/run/redis/redis.sock";
```

Comment:
```suggestion
        unixSocket = config.services.redis.unixSocket;
```

Reply:
Error build configuration:
```
building all machine configurations...
error: infinite recursion encountered, at /home/lf/works/src-nix/nixpkgs/lib/modules.nix:523:9
(use '--show-trace' to show detailed location information)

```

Reply:
By default `config.services.redis.unixSocket = null;`. Because of this, an error occurs. No need to change.